File: notification-application/src/main/java/com/example/notification/application/exception/ApplicationException.java
package com.example.notification.application.exception;

/** Application 계층 공통 예외 부모. adapter-in 에서 HTTP 매핑. */
public abstract class ApplicationException extends RuntimeException {

    protected ApplicationException(String message) {
        super(message);
    }

    protected ApplicationException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: notification-application/src/main/kotlin/com/example/notification/application/exception/ApplicationException.kt
package com.example.notification.application.exception

/** Application 계층 공통 예외 부모. adapter-in 에서 HTTP 매핑. */
abstract class ApplicationException : RuntimeException {
    protected constructor(message: String) : super(message)
    protected constructor(message: String, cause: Throwable?) : super(message, cause)
}
